src/rust/engine/options/src/env.rs

```diff
@@ -19,8 +19,29 @@ impl Env {
     Self { env }
   }
 
-  pub fn capture() -> Self {
-    Self::new(env::vars().collect::<HashMap<_, _>>())
+  pub fn capture_lossy() -> (Self, Vec<String>) {
+    let env_os = env::vars_os();
+    let mut env: HashMap<String, String> = HashMap::with_capacity(env_os.size_hint().0);
+    let mut dropped: Vec<String> = Vec::new();
+    for (os_key, os_val) in env_os {
+      match os_key.into_string() {
+        Ok(key) => match os_val.into_string() {
+          Ok(val) => {
+            env.insert(key, val);
+          }
+          Err(_) => {
+            dropped.push(key);
+          }
+        },
+        Err(os_key) => {
+          // We'll only be able to log the lossy name of any non-UTF-8 keys, but
+          // the user will know which one we mean.
+          dropped.push(os_key.to_string_lossy().to_string());
+        }
+      }
```

Comment:
FYI, just stylistically for Rust, this could expressed as a match on a tuple. This makes the `os_val.into_string()` happen more often, but the "hot"/common path is running both `into_string`s anyway, so I think that's fine... It also makes it further from the Python code 🤷‍♂️ 

```suggestion
      match (os_key.into_string(), os_val.into_string()) {
        (Ok(key), Ok(val)) => { env.insert(key, val); },
        (Ok(key), Err(_)) => dropped.push(key),
        // We'll only be able to log the lossy name of any non-UTF-8 keys, but
        // the user will know which one we mean.
        (Err(os_key), _) => dropped.push(os_key.to_string_lossy().to_string()),
      }
```

(I suspect that phrasing doesn't pass `cargo fmt`)